tests/test_termui.py

```diff
@@ -69,3 +70,42 @@ def test_secho(runner):
         click.secho(None, nl=False)
         bytes = out.getvalue()
         assert bytes == b''
+
+
+def test_progressbar_items_yields_all(runner):
+    expected = range(3)
+    with click.progressbar(expected) as progress:
+        assert list(progress.items()) == expected
+
+
+def test_progressbar_update(runner, monkeypatch):
+    class FakeClock(object):
+        def __init__(self):
+            self.now = time.time()
+
+        def advance_time(self, seconds=1):
+            self.now += seconds
+
+        def time(self):
+            return self.now
+
+    fake_clock = FakeClock()
+
+    @click.command()
+    def cli():
+        with click.progressbar(range(4)) as progress:
+            for _ in progress.items():
+                fake_clock.advance_time()
+                print("")
```

Comment:
Allows capture and assertion of the status bar update, after each iteration, by using _line splitting and "grep"_